# Add a "/" at the start of the topics to avoid namespacing
# The frame_ids are automatically namespaced in the rgbd.launch.py file.
# If the robot namespace is "robot_0" then the frame_ids become "robot_0/camera_link"

ORB_SLAM3_RGBD_ROS2:
  ros__parameters:
    # slam related
    robot_base_frame: base_footprint
    global_frame: map
    odom_frame: odom


    rgb_image_topic_name: /world/z_simple_palm_plantation/model/x500_custom_0/link/rgbd_link/sensor/rgbd/image
    depth_image_topic_name: /world/z_simple_palm_plantation/model/x500_custom_0/link/rgbd_link/sensor/rgbd/depth_image
    imu_topic_name: imu #Not used in this variant. Try RGB_IMU to use this information.

    # the below parameters are used to adjust the initial pose with respect to the map. 
    # the camera pose maybe translated and rotated with respect to the base_footprint. The wrapper publishes map->base_footprint. Hence this relative transform needs to be accounted.
    # the below transform is the transform from base_footpring -> camera_link
    robot_x: 0.0         # Gazebo <pose> X is 0
    robot_y: 0.0         # Gazebo <pose> Y is 0
    robot_z: 0.10        # Gazebo <pose> Z is 0.10 (10cm above ground, matches)
    robot_qx: 0.0        # No rotation around X, <pose> is 0
    robot_qy: 0.7071068  # sin(1.5707 / 2) ≈ 0.7071, 1.5707rad (90deg) pitch about Y (matches <pose>)
    robot_qz: 0.0        # No rotation around Z, <pose> is 0
    robot_qw: 0.7071068  # cos(1.5707 / 2) ≈ 0.7071

    visualization: true
    odometry_mode: false
    publish_tf: true
    map_data_publish_frequency: 1000 # publish every 1000.0 milliseconds
    do_loop_closing: true
